components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/WADLProcessor.java

```diff
@@ -488,40 +494,41 @@ private String saveSchema(String schemaUrl, String version, Properties props) th
 	 * @return                      Endpoint Path.
 	 * @throws RegistryException    If fails to create endpoint element.
 	 */
-	private String createEndpointElement(RequestContext requestContext, OMElement wadlElement, String version)
-			throws RegistryException {
-		OMNamespace wadlNamespace = wadlElement.getNamespace();
-		String wadlNamespaceURI = wadlNamespace.getNamespaceURI();
-		String wadlNamespacePrefix = wadlNamespace.getPrefix();
-		OMElement resourcesElement =
-				wadlElement.getFirstChildWithName(new QName(wadlNamespaceURI, "resources", wadlNamespacePrefix));
-		if (resourcesElement != null) {
-			String endpointUrl = resourcesElement.getAttributeValue(new QName("base"));
-			if (!StringUtils.isBlank(endpointUrl)) {
-				String endpointPath = EndpointUtils.deriveEndpointFromUrl(endpointUrl);
-				String endpointName = EndpointUtils.deriveEndpointNameWithNamespaceFromUrl(endpointUrl);
-				String endpointContent =
-						EndpointUtils.getEndpointContentWithOverview(endpointUrl, endpointPath, endpointName, version);
-				OMElement endpointElement;
-				try {
-					endpointElement = AXIOMUtil.stringToOM(endpointContent);
-				} catch (XMLStreamException e) {
-					throw new RegistryException("Error in creating the endpoint element. ", e);
-				}
-
-				return RESTServiceUtils.addEndpointToRegistry(requestContext, endpointElement, endpointPath);
-			} else {
-				log.warn("Base path does not exist. endpoint creation may fail. ");
-			}
-		} else {
-			log.warn("Resources element is null. ");
-		}
-		return null;
-	}
+    private String createEndpointElement(RequestContext requestContext, OMElement wadlElement, String version,
+            String servicePath) throws RegistryException {
```

Comment:
@Niranjan-K Thanks for pointing out. I have updated the JavaDocs with a description of a new parameter `servicePath` in #218 (a94e157e3554f10f57202245dd9b5701f0d5cc1b)